hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractRecordStore.java

```diff
@@ -83,12 +86,37 @@ protected AbstractRecordStore(MapContainer mapContainer, int partitionId) {
         this.mapStoreContext = mapContainer.getMapStoreContext();
         this.mapDataStore = mapStoreContext.getMapStoreManager().getMapDataStore(name, partitionId);
         this.lockStore = createLockStore();
-        Collection<RecordStoreMutationObserver<Record>> mutationObservers = mapServiceContext
-                .createRecordStoreMutationObservers(getName(), partitionId);
-        this.mutationObserver = new CompositeRecordStoreMutationObserver<>(mutationObservers);
+        this.mutationObserver = new CompositeMutationObserver<>();
         this.storeAdapter = new RecordStoreAdapter(this);
     }
 
+    @Override
+    public void init() {
+        this.storage = createStorage(recordFactory, inMemoryFormat);
+        addMutationObservers();
+    }
+
+    // Overridden in EE.
+    protected void addMutationObservers() {
+        // Add observer for event journal
+        EventJournalConfig eventJournalConfig = mapContainer.getEventJournalConfig();
+        if (eventJournalConfig != null && eventJournalConfig.isEnabled()) {
+            mutationObserver.add(new EventJournalWriterMutationObserver(mapServiceContext.getEventJournal(),
+                    mapContainer, partitionId));
+        }
+
+        // Add observer for json metadata if map is not NATIVE
```

Comment:
Minor: it's added even if it's `NATIVE` (by the overridden method).

Reply:
done